eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/views/report/details/BugPathContentProvider.java

```diff
@@ -24,7 +27,13 @@ public void inputChanged(Viewer viewer, Object oldInput, Object newInput) {
             @SuppressWarnings("unchecked") Optional<ProblemInfo> bp = (Optional<ProblemInfo>)
                     inputElement;
             if (bp.isPresent()) {
-                return bp.get().getItems().toArray();
+                ArrayList<BugPathItem> result = new ArrayList<>(bp.get().getItems().asList());
+                for (int i = result.size() - 1; i >= 0; --i) {
+                    if (result.get(i).getMessage() == "") {
+                        result.remove(i);
+                    }
+                }
+                return result.toArray();
```

Comment:
Guava contains a function named [removeIf](https://google.github.io/guava/releases/snapshot/api/docs/com/google/common/collect/Iterables.html#removeIf(java.lang.Iterable,%20com.google.common.base.Predicate)) for exactly this loop.